src/coinbase/address/wallet_address.ts

```diff
@@ -340,4 +352,251 @@ export class WalletAddress extends Address {
+    options: { [key: string]: string } = {},
+    timeoutSeconds = 60,
+    intervalSeconds = 0.2,
+  ): Promise<StakingOperation> {
+    await this.validateCanUnstake(amount, assetId, mode, options);
+    return this.createStakingOperation(
+      amount,
+      assetId,
+      "unstake",
+      mode,
+      options,
+      timeoutSeconds,
+      intervalSeconds,
+    );
+  }
+
+  /**
+   * Creates a staking operation to claim stake.
+   *
+   * @param amount - The amount to claim stake.
+   * @param assetId - The asset to claim stake.
+   * @param mode - The staking mode. Defaults to DEFAULT.
+   * @param options - Additional options such as setting the mode for the staking action.
+   * @param timeoutSeconds - The amount to wait for the transaction to complete when broadcasted.
+   * @param intervalSeconds - The amount to check each time for a successful broadcast.
+   * @returns The staking operation after it's completed successfully.
+   */
+  public async createClaimStake(
+    amount: Amount,
+    assetId: string,
+    mode: StakeOptionsMode = StakeOptionsMode.DEFAULT,
+    options: { [key: string]: string } = {},
+    timeoutSeconds = 60,
+    intervalSeconds = 0.2,
+  ): Promise<StakingOperation> {
+    await this.validateCanClaimStake(amount, assetId, mode, options);
+    return this.createStakingOperation(
+      amount,
+      assetId,
+      "claim_stake",
+      mode,
+      options,
+      timeoutSeconds,
+      intervalSeconds,
+    );
+  }
+
+  /**
+   * Creates a staking operation to stake, signs it, and broadcasts it on the blockchain.
+   *
+   * @param amount - The amount for the staking operation.
+   * @param assetId - The asset to the staking operation.
+   * @param action - The type of staking action to perform.
+   * @param mode - The staking mode. Defaults to DEFAULT.
+   * @param options - Additional options such as setting the mode for the staking action.
+   * @param timeoutSeconds - The amount to wait for the transaction to complete when broadcasted.
+   * @param intervalSeconds - The amount to check each time for a successful broadcast.
+   * @returns The staking operation after it's completed fully.
+   */
+  private async createStakingOperation(
+    amount: Amount,
+    assetId: string,
+    action: string,
+    mode: StakeOptionsMode,
+    options: { [key: string]: string },
+    timeoutSeconds: number,
+    intervalSeconds: number,
+  ): Promise<StakingOperation> {
+    let stakingOperation = await this.createStakingOperationRequest(
+      amount,
+      assetId,
+      action,
+      mode,
+      options,
+    );
+
+    // NOTE: Staking does not yet support server signers at this point.
+    await stakingOperation.sign(this.key!);
+    for (let i = 0; i < stakingOperation.getTransactions().length; i++) {
+      const transaction = stakingOperation.getTransactions()[0];
+      if (!transaction.isSigned()) {
+        continue;
+      }
+      stakingOperation = await this.broadcastStakingOperationRequest(
```

Comment:
do we need to track if a tx was previously broadcasted - if yes - then ignore ? Don't see any such handling within `broadcastStakingOperationRequest`

Reply:
There is a check to see if the transaction is signed above this call. The assumption is that if it's signed id' be broadcasted already. Would this break the dedicated ETH flow?

Reply:
Got it - Make sense. I think let's roll ahead with this and I can see if something needs to be addressed later for dedicated eth staking.